### ISSUE TITLE
Integration With Plots2

### ISSUE THREAD

**Issue by IshaGupta18** [publiclab/simple-data-grapher#60]
This is the next thing we need to do.
Once this is integrated with the main site like PL editor, the functionality will remain the same, but we need to understand how this is to be done.
I am not very sure about how to begin with it, and it would be great if I could get some guidance here for the same.
Once we have integrated this, we have to implement (some or all) of the following features:

- [ ] Displaying per-user data

- [ ] Publishing generated charts as a `Research Note`

- [ ] Plot charts from previously uploaded files.

Detailed explanations can be found at https://publiclab.org/notes/IshaGupta18/03-04-2019/gsoc-proposal-sensor-data-upload-and-display-library

@jywarren @namangupta01 @gauravano @rexagod @IgorWilbert it would be great if you all could start me up with this one. Thanks a ton!

**Comment by jywarren** [comment]
Hi Isha! Just copying in the note from the open call on this:


- example of current CSV graph integration: https://github.com/publiclab/plots2/blob/master/app/views/grids/_graph.html.erb, this is generated when a graph powertag is used on the site: https://publiclab.org/wiki/power-tags#Inline+graphs

See how this code pattern-matches the `[graph:example.csv]` in node and comment text, and replaces it with the `_graph.html.erb` template?

https://github.com/publiclab/plots2/blob/61684ae2fce32b5c53fba23cb7be46dd69867b60/app/models/concerns/node_shared.rb#L38-L54

**Comment by jywarren** [comment]
So let's implement a second version like `[simple-graph:/path/to/example.csv]` by copying that filter function and then adding a second html template.

**Comment by jywarren** [comment]
and once we feel good about it we can replace the original!

**Comment by IshaGupta18** [comment]
Oh alright, let me have a look and understand these ones! Thank you!

**Comment by IshaGupta18** [comment]
> So let's implement a second version like `[simple-graph:/path/to/example.csv]` by copying that filter function and then adding a second html template.

I am not very sure on what exactly we are supposed to do here? Could you please elaborate more on this?

**Comment by IshaGupta18** [comment]
@jywarren I am happy to tell you that I have published the first version of simple-data-grapher on npm and will be updating it soon to remove redundant code and I am about to begin with the process of integrating it with plots2 very soon.

https://www.npmjs.com/package/simple-data-grapher

**Comment by IshaGupta18** [comment]
@jywarren I have done the basic integration with plots2!!
![image](https://user-images.githubusercontent.com/40794215/60500530-22123b80-9cd8-11e9-913e-055726df7448.png)

**Comment by jywarren** [comment]
oh wow, cool - can you share a PR? How does it work?

**Comment by IshaGupta18** [comment]
Yes definitely, I would be creating one in plots2 and share the link here!

**Comment by jywarren** [comment]
This is a simpler example we mentioned on the call!

https://github.com/publiclab/plots2/blob/master/app/views/editor/rich.html.erb#L17-L18

**Comment by jywarren** [comment]
Here is where we're inserting the Ruby values into the constructor of the Editor library! 

https://github.com/publiclab/plots2/blob/a9d4e3048eee72aa3161f8999a6fde5b95443e58/app/views/editor/rich.html.erb#L315-L342

**Comment by IshaGupta18** [comment]
Here's the PR @jywarren https://github.com/publiclab/plots2/pull/5993
